### PR TITLE
add LeViT

### PR DESCRIPTION
**Pull request by kaijieshi7**
## Add LeViT
- [x] build model
- [x] update __init__.py in models
- [x] convert pretrained weight
- [x] inference test on imagenet and update model_zoo
- [x] update docs
- [x] update readme
- [x] update changelog
- [x] pytorch speed comparison

### CODE REVIEW

**Comment by kaijieshi7** [comment]
OneFlow LeViT_128S time: 34.5ms (= 3447.5ms / 100, input_shape=[16, 3, 224, 224])
PyTorch LeViT_128S time: 40.7ms (= 4071.0ms / 100, input_shape=[16, 3, 224, 224])
✔️ Relative speed: 1.18 (= 40.7ms / 34.5ms)

OneFlow LeViT_128 time: 45.5ms (= 4551.8ms / 100, input_shape=[16, 3, 224, 224])
PyTorch LeViT_128 time: 45.8ms (= 4576.1ms / 100, input_shape=[16, 3, 224, 224])
✔️ Relative speed: 1.01 (= 45.8ms / 45.5ms)

OneFlow LeViT_192 time: 56.6ms (= 5663.3ms / 100, input_shape=[16, 3, 224, 224])
PyTorch LeViT_192 time: 49.0ms (= 4904.7ms / 100, input_shape=[16, 3, 224, 224])
❌ Relative speed: 0.87 (= 49.0ms / 56.6ms)

OneFlow LeViT_256 time: 60.8ms (= 6081.3ms / 100, input_shape=[16, 3, 224, 224])
PyTorch LeViT_256 time: 49.4ms (= 4942.2ms / 100, input_shape=[16, 3, 224, 224])
❌ Relative speed: 0.81 (= 49.4ms / 60.8ms)

OneFlow LeViT_384 time: 89.8ms (= 8976.2ms / 100, input_shape=[16, 3, 224, 224])
PyTorch LeViT_384 time: 55.5ms (= 5551.0ms / 100, input_shape=[16, 3, 224, 224])
❌ Relative speed: 0.62 (= 55.5ms / 89.8ms)

OneFlow LeViT_128S time: 15.2ms (= 1517.8ms / 100, input_shape=[16, 3, 224, 224], backward is disabled, ddp, world size=2)
PyTorch LeViT_128S time: 9.7ms (= 967.1ms / 100, input_shape=[16, 3, 224, 224], backward is disabled, ddp, world size=2)
❌ Relative speed: 0.64 (= 9.7ms / 15.2ms)
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************

OneFlow LeViT_128 time: 15.4ms (= 1544.3ms / 100, input_shape=[16, 3, 224, 224], backward is disabled, ddp, world size=2)
PyTorch LeViT_128 time: 10.1ms (= 1011.0ms / 100, input_shape=[16, 3, 224, 224], backward is disabled, ddp, world size=2)
❌ Relative speed: 0.65 (= 10.1ms / 15.4ms)
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************

OneFlow LeViT_192 time: 19.2ms (= 1918.2ms / 100, input_shape=[16, 3, 224, 224], backward is disabled, ddp, world size=2)
PyTorch LeViT_192 time: 9.8ms (= 975.5ms / 100, input_shape=[16, 3, 224, 224], backward is disabled, ddp, world size=2)
❌ Relative speed: 0.51 (= 9.8ms / 19.2ms)
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************

OneFlow LeViT_256 time: 22.5ms (= 2250.0ms / 100, input_shape=[16, 3, 224, 224], backward is disabled, ddp, world size=2)
PyTorch LeViT_256 time: 12.2ms (= 1216.7ms / 100, input_shape=[16, 3, 224, 224], backward is disabled, ddp, world size=2)
❌ Relative speed: 0.54 (= 12.2ms / 22.5ms)
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************

OneFlow LeViT_384 time: 31.9ms (= 3191.4ms / 100, input_shape=[16, 3, 224, 224], backward is disabled, ddp, world size=2)
PyTorch LeViT_384 time: 11.8ms (= 1177.6ms / 100, input_shape=[16, 3, 224, 224], backward is disabled, ddp, world size=2)
❌ Relative speed: 0.37 (= 11.8ms / 31.9ms)
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************

**Comment by kaijieshi7** [comment]
这个速度误差好大啊，大家有没有什么想法

**Comment by rentainhe** [comment]
> 这个速度误差好大啊，大家有没有什么想法

我看看，你搬的速度有点快233

File: README.md [hunk 1]
```diff
@@ -82,6 +82,7 @@ pip install flowvision==0.1.0
             <li>RegionViT</li>
             <li>UniFormer</li>
             <li>VAN</li>
+            <li>LeViT</li>
```
Comment (rentainhe) [inline README.md:85]: LeViT可以放到前面一些，RegionViT之前的文章了~

File: docs/source/flowvision.models.rst [hunk 2]
```diff
@@ -42,6 +42,7 @@ architectures for image classification:
 -  `ConvNeXt`_
 -  `RegionViT`_
 -  `VAN`_
+_  `LeViT`_
```
Comment (rentainhe) [inline docs/source/flowvision.models.rst:45]: 同理，可以放到RegionViT之前

File: flowvision/models/levit.py [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+
+class Residual(flow.nn.Module):
+    def __init__(self, m, drop):
+        super().__init__()
+        self.m = m
+        self.drop = drop
+
+    def forward(self, x):
+        if self.training and self.drop > 0:
+            return x + self.m(x) * (flow.rand(x.size(0), 1, 1, device=x.device)>self.drop).div(1 - self.drop).detach()
```
Comment (rentainhe) [inline flowvision/models/levit.py:80]: `.div`的速度可能有问题

File: flowvision/models/levit.py [hunk 4]
```diff
@@ -0,0 +1,492 @@
...
+
+class Subsample(flow.nn.Module):
+    def __init__(self, stride, resolution):
+        super().__init__()
+        self.stride = stride
+        self.resolution = resolution
+
+    def forward(self, x):
+        B, N, C = x.shape
+        x = x.view(B, self.resolution, self.resolution, C)[
```
Comment (rentainhe) [inline flowvision/models/levit.py:154]: `.view`还有一些slice的操作速度上可能也有差距

File: results/results_imagenet.md [hunk 5]
```diff
@@ -178,6 +178,11 @@
 | van_small | 81.012 | 18.988 | 95.626 | 4.374 | 13.9M | 224 | 0.9 | bicubic |
 | van_base  | 82.806 | 17.194 | 96.206 | 3.794 | 26.6M | 224 | 0.9 | bicubic |
 | van_large | 83.860 | 16.140 | 96.728 | 3.272 | 44.8M | 224 | 0.9 | bicubic |
+| LeViT-128S | 76.518 | 23.482 | 92.904 | 7.096 | 7.8M  | 224 | 0.875 | bicubic |
```
Comment (rentainhe) [inline results/results_imagenet.md:181]: 这里的命名改为`levit_128s`

File: flowvision/models/levit.py [hunk 6]
```diff
@@ -0,0 +1,492 @@
...
+
+        >>> import flowvision
+        >>> LeViT_256 = flowvision.models.LeViT_256(pretrained=False, progress=True)
+
+    """
+    return model_factory(**specification['LeViT_256'], num_classes=num_classes,
+                         distillation=distillation, pretrained=pretrained,)
+
+@ModelCreator.register_model
+def LeViT_384(num_classes=1000, distillation=True,
```
Comment (rentainhe) [inline flowvision/models/levit.py:471]: 这里还是改为全小写吧，记得model_zoo下也要改一下, `levit_384`

File: results/results_imagenet.md (hunk 5 above)
Comment (rentainhe) [reply results/results_imagenet.md:181]: 然后可以放到RegionViT之前~

File: README.md (hunk 1 above)
Comment (kaijieshi7) [reply README.md:85]: 已改

File: docs/source/flowvision.models.rst (hunk 2 above)
Comment (kaijieshi7) [reply docs/source/flowvision.models.rst:45]: 已改

File: results/results_imagenet.md (hunk 5 above)
Comment (kaijieshi7) [reply results/results_imagenet.md:181]: 已改

File: flowvision/models/levit.py (hunk 6 above)
Comment (kaijieshi7) [reply flowvision/models/levit.py:471]: 已改

**Review by rentainhe — APPROVED** [review]

**Comment by rentainhe** [comment]
修复一下冲突就可以合并了 @kaijieshi7

**Comment by kaijieshi7** [comment]
> 修复一下冲突就可以合并了 @kaijieshi7

OK

**Pull request merged**